src/lib/repositories/impl_v2/oauth2_repository_impl.py

```diff
@@ -104,8 +104,9 @@ def validate_token(self, token):
         # That is the way how jwt library validates the token
         jwt.decode(token, self.env_config.oauth2_secret_key, algorithms="HS256")
 
-    def refresh_token(self, refresh_token):
+    def refresh_token(self, refresh_token, client_id, client_secret):
         self.validate_token(refresh_token)
+        self._validate_client_credentials(client_id, client_secret)
 
         app_refresh_token = self._get_refresh_token_by_token(refresh_token)
```

Comment:
It is better to make sure data is consistent, so try to look for this refresh token using the access toke as well.

Reply:
Done.